File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/OrgRole.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum OrgRole {
    Owner,
    Admin,
    Member;

    public String toWire() {
        return switch (this) {
            case Owner -> "owner";
            case Admin -> "admin";
            case Member -> "member";
        };
    }

    public static OrgRole fromWire(String value) {
        if (value == null || value.isBlank()) throw new IllegalArgumentException("OrgRole is required.");
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "owner" -> Owner;
            case "admin" -> Admin;
            case "member" -> Member;
            default -> throw new IllegalArgumentException("Unknown OrgRole: " + value);
        };
    }
}

File: apps/backend/src/Common/Domain/ValueObjects/OrgRole.cs
namespace OffceOs.Domain.Common.ValueObjects;

public enum OrgRole
{
    Owner,
    Admin,
    Editor,
    Viewer,
}

public static class OrgRoleExtensions
{
    public static string ToStorageString(this OrgRole role) => role switch
    {
        OrgRole.Owner => "Owner",
        OrgRole.Admin => "Admin",
        OrgRole.Editor => "Editor",
        OrgRole.Viewer => "Viewer",
        _ => throw new ArgumentOutOfRangeException(nameof(role)),
    };

    public static OrgRole ToOrgRole(this string value) => value switch
    {
        "Owner" => OrgRole.Owner,
        "Admin" => OrgRole.Admin,
        "Member" => OrgRole.Editor,
        "Editor" => OrgRole.Editor,
        "Viewer" => OrgRole.Viewer,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown org role: {value}"),
    };
}
